src/command/dev/next/router/install.rs

```diff
@@ -0,0 +1,251 @@
+use apollo_federation_types::config::RouterVersion;
+use async_trait::async_trait;
+use camino::{Utf8Path, Utf8PathBuf};
+use semver::Version;
+
+use crate::{
+    command::{install::Plugin, Install},
+    options::LicenseAccepter,
+    utils::{client::StudioClientConfig, effect::install::InstallBinary},
+};
+
+use super::binary::RouterBinary;
+
+#[derive(thiserror::Error, Debug)]
+#[error("Failed to install the router")]
+pub enum InstallRouterError {
+    #[error("unable to find dependency: \"{err}\"")]
+    MissingDependency {
+        /// The error while attempting to find the dependency
+        err: String,
+    },
+    #[error("Missing filename for path: {path}")]
+    MissingFilename { path: Utf8PathBuf },
+    #[error("Invalid semver version: \"{input}\"")]
+    Semver {
+        input: String,
+        source: semver::Error,
+    },
+}
+
+pub struct InstallRouter {
+    studio_client_config: StudioClientConfig,
+    router_version: RouterVersion,
+}
+
+impl InstallRouter {
+    #[allow(unused)]
+    pub fn new(
+        router_version: RouterVersion,
+        studio_client_config: StudioClientConfig,
+    ) -> InstallRouter {
+        InstallRouter {
+            router_version,
+            studio_client_config,
+        }
+    }
+}
+
+#[async_trait]
+impl InstallBinary for InstallRouter {
+    type Binary = RouterBinary;
+    type Error = InstallRouterError;
+    async fn install(
+        &self,
+        override_install_path: Option<Utf8PathBuf>,
+        elv2_license_accepter: LicenseAccepter,
+        skip_update: bool,
+    ) -> Result<Self::Binary, Self::Error> {
```

Comment:
doublechecking: `override_install_path` and `skip_update` are the only cli options we need to support? that's my read of the help for `dev`

Reply:
That's my read as well.